packages/html/src/events/form.rs

```diff
@@ -23,6 +30,52 @@ pub struct FormData {
     pub files: Option<std::sync::Arc<dyn FileEngine>>,
 }
 
+impl FormData {
+    // ***** function to parse the 'values' to make it ready to use*******
+    // e.g - self.values = { username: ["rust"], password: ["dioxus"]}
+    // what we need it to be: { username: "rust", password: "dioxus"}
+    fn get_parsed_values(&self) -> Option<HashMap<String, ValueType>> {
+        if self.values.is_empty() {
+            return None;
+        }
+
+        let raw_values = self.values.clone();
+
+        let mut parsed_values: HashMap<String, ValueType> = HashMap::new();
+
+        for (fieldname, values) in raw_values.into_iter() {
+            /*
+            case 1 - multiple values, example { driving_types: ["manual", "automatic"] }
+                In this case we want to return the values as it is, NO point in making
+                driving_types: "manual, automatic"
+
+            case 2 - single value, example { favourite_language: ["rust"] }
+                In this case we would want to deserialize the value as follows
+                favourite_language: "rust"
+            */
+            if values.len() > 1 {
```

Comment:
Much better! This still leaves one situation where this will deserialize incorrectly:
If I have rsx like this and I only select the first option:
```rust
form {
    onsubmit: move |evt| {
        // How should we deserialize the form data?
        println!("{:#?}", evt.values);
    },
    select {
        multiple: "true",
        name: "options",

        option {
            value: "option1",
            "The first option"
        }
        option {
            value: "option2",
            "The second option"
        }
        option {
            value: "option3",
            "The third option"
        }
    }
}
```
And I want to deserialize it into this struct:
```rust
#[derive(Deserialize)]
enum MyOptions {
    #[serde(rename = "option1")]
    Option1,
    #[serde(rename = "option2")]
    Option2,
    #[serde(rename = "option3")]
    Option3,
}

#[derive(Deserialize)]
struct FormData {
    options: Vec<MyOptions>,
}
```

We can't know from the data we currently have in the FormData struct if the data is from a single value or from many values.

This is the issue I was talking about here:
> We could either add something to the form data to indicate that a value was a list or we could modify the deserializer to try to deserialize a single value list as either a single value or a list.

We could [write a custom data format](https://serde.rs/data-format.html) or just change the type of values to `Hashmap<String, ValueType>` and serialize the data correctly when we send the event from javascript

Reply:
Yes, that will be a problem. It would be better if I get the type of values and use it in the if condition. I'll check if I can get a grab on that.